rwslib/__init__.py

```diff
@@ -73,45 +86,52 @@ def send_request(self, request_object, timeout=None, retries=1, **kwargs):
         # Construct a URL from the object and make a call
         full_url = make_url(self.base_url, request_object.url_path())
         if request_object.requires_authorization:
-            kwargs['auth'] = self.auth
+            kwargs["auth"] = self.auth
             # TODO: Look at different connect and read timeouts?
-            kwargs['timeout'] = timeout
+            kwargs["timeout"] = timeout
             kwargs.update(request_object.args())
 
-
-        #Explicit use of requests library here. Could alter in future to inject library to use in case
-        #requests not available.
+        # Explicit use of requests library here. Could alter in future to inject library to use in case
+        # requests not available.
 
         # Get a session that allows us to customize HTTP requests
         session = requests.Session()
 
         # Mount a custom adapter that retries failed connections for HTTP and HTTPS requests.
-        for scheme in ["http://","https://"]:
+        for scheme in ["http://", "https://"]:
             session.mount(scheme, requests.adapters.HTTPAdapter(max_retries=retries))
 
-
-        action = {"GET": session.get,
-                  "POST": session.post}[request_object.method]
+        action = {"GET": session.get, "POST": session.post}[request_object.method]
 
         start_time = time.time()
 
         try:
-            r = action(full_url, **kwargs)
-        except (requests.exceptions.ConnectTimeout, requests.exceptions.ReadTimeout) as exc:
+            r = action(full_url, **kwargs)  # type: requests.models.Response
+        except (
+            requests.exceptions.ConnectTimeout,
+            requests.exceptions.ReadTimeout,
+        ) as exc:
             if isinstance(exc, (requests.exceptions.ConnectTimeout,)):
-                raise RWSException("Server Connection Timeout", "Connection timeout for {}".format(full_url))
+                raise RWSException(
+                    "Server Connection Timeout",
+                    "Connection timeout for {}".format(full_url),
+                )
             elif isinstance(exc, (requests.exceptions.ReadTimeout,)):
-                raise RWSException("Server Read Timeout", "Read timeout for {}".format(full_url))
+                raise RWSException(
+                    "Server Read Timeout", "Read timeout for {}".format(full_url)
+                )
 
         self.request_time = time.time() - start_time
-        self.last_result = r #see also r.elapsed for timedelta object.
+        self.last_result = r  # see also r.elapsed for timedelta object.
 
+        # ensure the response.text is usable
+        r.encoding = "utf-8-sig"
```

Comment:
Here we force the encoding to utf-8-sig. Does RWS *always* return text utf-8 encoded with BOM? Should it be up to the request type to set this instead? What if you have a custom dataset?

Reply:
Requests does the magic here!